llama-cpp-2/src/token/data_array.rs

```diff
@@ -1,18 +1,24 @@
 //! an rusty equivalent of `llama_token_data`.
-use crate::token::data::LlamaTokenData;
+use std::{ffi::CString, ptr};
+
+use crate::{model::LlamaModel, token::data::LlamaTokenData};
+
+use super::LlamaToken;
 
 /// a safe wrapper around `llama_token_data_array`.
 #[derive(Debug, Clone, PartialEq)]
 #[allow(clippy::module_name_repetitions)]
 pub struct LlamaTokenDataArray {
     /// the underlying data
     pub data: Vec<LlamaTokenData>,
+    /// the selected token
+    pub selected: i64,
```

Comment:
I think I would prefer a Option<u32> or something similar. I would prefer not to use -1 as a sentinel in a pub struct like this. (are other negative values valid?)

Reply:
It is now an `Option<usize>`, which should be much clearer as to its purpose and easier to use.